conf/modules.config

```diff
@@ -20,164 +19,117 @@ process {
-        // Argument list needs to end with --peptides
-        ext.args   = [
-                    genome_reference != 'grch37' & genome_reference != 'grch38' ? "--genome_reference '${genome_reference}'" : '',
-                    genome_reference == 'grch37'                                ? "--genome_reference 'https://grch37.ensembl.org/'" : '',
-                    genome_reference == 'grch38'                                ? "--genome_reference 'https://www.ensembl.org'" : '',
-                    '--peptides'
-        ].join(' ').trim()
+    withName: CAT_FASTA {
+        ext.prefix = {"${meta.sample}"}
         publishDir = [
-            path: { "${params.outdir}/split_predictions/${meta.sample}" },
-            mode: params.publish_dir_mode
+            path: { "${params.outdir}/epytope" },
+            mode: params.publish_dir_mode,
+            pattern: '*.fasta'
         ]
     }
 
-    withName: EPYTOPE_PEPTIDE_PREDICTION_VAR {
-        // Argument list needs to end with --somatic_mutation
-        ext.args   = [
-                    genome_reference != 'grch37' & genome_reference != 'grch38' ? "--genome_reference '${genome_reference}'" : '',
-                    genome_reference == 'grch37'                                ? "--genome_reference 'https://grch37.ensembl.org/'" : '',
-                    genome_reference == 'grch38'                                ? "--genome_reference 'https://www.ensembl.org'" : '',
-                    '--somatic_mutation'
-        ].join(' ').trim()
-        publishDir = [
-            path: { "${params.outdir}/split_predictions/${meta.sample}" },
-            mode: params.publish_dir_mode
-        ]
+    withName: FASTA2PEPTIDES {
+        publishDir = [ enabled: false ]
     }
 
-    withName: MERGE_JSON_SINGLE {
-        ext.args = " --single_input "
+    withName: SPLIT_PEPTIDES {
+        publishDir = [ enabled: false ]
     }
 
-    withName: MERGE_JSON_MULTI {
-        ext.args = " --input \$PWD "
+    withName: PREPARE_PREDICTION_INPUT {
+        ext.prefix = {"${meta.file_id}"}
+        publishDir = [ enabled: false ]
     }
 
-    withName: CAT_TSV {
+    withName: MHCFLURRY {
+        ext.prefix = {"${meta.file_id}"}
         publishDir = [
-            path: { "${params.outdir}/predictions/${meta.sample}" },
-            mode: params.publish_dir_mode
+            path: { "${params.outdir}/mhcflurry" },
+            mode: params.publish_dir_mode,
+            pattern: '*.csv'
         ]
     }
 
-    withName: CAT_FASTA {
+    withName: MHCNUGGETS {
+        ext.prefix = {"${meta.file_id}"}
         publishDir = [
-            path: { "${params.outdir}/predictions/${meta.sample}" },
-            mode: params.publish_dir_mode
+            path: { "${params.outdir}/mhcnuggets" },
+            mode: params.publish_dir_mode,
+            pattern: '*predicted_mhcnuggets.csv'
         ]
     }
 
-    withName: CSVTK_CONCAT {
+    withName: NETMHCPAN {
```

Comment:
Can you combine the withNames of NETMHCPAN and NETMHCIIPAN, they have the same parameters

Reply:
But you can assign this that the folder is assigned in the name of the process right?
Then you can combine this information (it is also something minor)

Reply:
I think its cleaner to keep them separate, bc NetMHCIIpan can have slightly different additional CLI params comparted to NetMHCpan 🤔

Reply:
Then it is fine to keep them separate, I just dont like cp-ed code 😄